tools/src/main/scala/scala/scalanative/build/Filter.scala

```diff
@@ -33,44 +45,83 @@ private[scalanative] object Filter {
   ): (Seq[Path], Config) = {
     val nativeCodePath = destPath.resolve(nativeCodeDir)
     // check if filtering is needed, o.w. return all paths
-    findFilterProperties(nativeCodePath).fold((allPaths, config)) { file =>
-      // predicate to check if given file path shall be compiled
-      // we only include sources to the base of the gc code and exclude
-      // all optional dependencies if they are not necessary
-      val optPath = nativeCodePath.resolve("optional").abs
-      val gcPath = nativeCodePath.resolve("gc").abs
-
-      def include(path: String) = {
-        if (path.contains(optPath)) {
-          val name = Paths.get(path).toFile.getName.split("\\.").head
-          linkerResult.links.exists(_.name == name)
-        } else {
-          true
+    findFilterProperties(nativeCodePath).fold((allPaths, config)) { filepath =>
+
+      val desc =
+        tl.Using(new FileReader(filepath.toFile())) { reader =>
+          val props = new Properties()
+          props.load(reader)
+          Descriptor(
+            Option(props.getProperty("project.groupId")),
+            Option(props.getProperty("project.artifactId")),
+            Option(props.getProperty("nir.link.names")) match {
+              case Some(value) => value.split(',').map(_.trim()).toList
+              case None        => List.empty
+            }
+          )
```

Comment:
How about refactoring it out to helper function`Descriptor.load(Path): Try[Descriptor]`?

Reply:
This is a very good idea - looks good.

What do you think about the prefix being `SCALANATIVE_LINK_` as suggested by Lee?

Reply:
It's a good improvement in my opinion. Personally I'd even like `ScalaNative_if_links_x` but that might be too verbose. `ScalaNative_Link_x` seems to have good balance between verbosity and name clash prevention.

Reply:
I think because there is already `#ifdef` or `#if defined()` the one with out `if` is ok.